rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -110,64 +182,135 @@ fn test_get_inputs() {
+    \x36\x4b\x3e\xdf\x84\xb5\x20\xa0\x18\xd5\x7e\x71\xfa\xce\x19\x7e\xc8\xf9\x46\x43\x60\x7e\x4a\xca\
+    \x70\xdc\x82\xc1\xfd\xff\xff\xff\x02\x10\x27\x00\x00\x00\x00\x00\x00\x19\x76\xa9\x14\x62\xe9\x07\
+    \xb1\x5c\xbf\x27\xd5\x42\x53\x99\xeb\xf6\xf0\xfb\x50\xeb\xb8\x8f\x18\x88\xac\x00\x96\x00\x00\x00\
+    \x00\x00\x00\x19\x76\xa9\x14\xb9\x73\x68\xd8\xbf\x0a\x37\x69\x00\x85\x16\x57\xf3\x7f\xbe\x73\xa6\
+    \x56\x61\x33\x88\xac\xb3\xa3\x0c\x00"
+                    .to_vec(),
+            }),
+        });
 
     let result = env
         .await_call(call_id)
         .expect("the fetch request didn't finish");
 
     match &result {
         WasmResult::Reply(bytes) => {
-            let response = Decode!(bytes, Vec<String>).unwrap();
-            assert_eq!(
-                response,
-                vec!["1HuaGHYa4rjqpcqkm1thE48s88rf1QpR5J".to_string()],
-            );
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(response, CheckTransactionResponse::Passed));
         }
         WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
     }
+
+    let cycles_after = env.cycle_balance(setup.universal_canister);
+    let expected_cost =
+        CHECK_TRANSACTION_CYCLES_SERVICE_FEE + 2 * get_tx_cycle_cost(INITIAL_BUFFER_SIZE);
+    let actual_cost = cycles_before - cycles_after;
+    // Since we use universal_canister, actual_cost is greater than expected_cost
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < 7_000_000);
+}
+
+#[test]
+fn test_check_transaction_error() {
+    let setup = setup_env();
+    let cycles_before = setup.env.cycle_balance(setup.universal_canister);
+    let mut txid =
+        Txid::from_str("a80763842edc9a697a2114517cf0c138c5403a761ef63cfad1fa6993fa3475ed")
+            .unwrap()
+            .as_ref()
+            .to_vec();
+
+    // Test for cycles not enough
+    let call_id = submit_universal_call(
+        &setup,
+        setup.kyt_canister,
+        "check_transaction",
+        Encode!(&CheckTransactionArgs { txid: txid.clone() }).unwrap(),
+        CHECK_TRANSACTION_CYCLES_REQUIRED - 1,
+    )
+    .expect("submit_call failed to return call id");
+    let result = setup
+        .env
+        .await_call(call_id)
+        .expect("the fetch request didn't finish");
+    match &result {
+        WasmResult::Reply(bytes) => {
+            let response = Decode!(bytes, CheckTransactionResponse).unwrap();
+            assert!(matches!(
+                response,
+                CheckTransactionResponse::NotEnoughCycles
+            ));
+        }
+        WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
+    };
+    let cycles_after = setup.env.cycle_balance(setup.universal_canister);
+    let expected_cost = CHECK_TRANSACTION_CYCLES_SERVICE_FEE;
+    let actual_cost = cycles_before - cycles_after;
+    // Since we use universal_canister, actual_cost is greater than expected_cost
```

Comment:
Why?

Reply:
More comments are added to explain this.